frame/rewards/src/lib.rs

```diff
@@ -182,40 +247,32 @@ decl_module! {
 			Self::do_update_locks(&target, locks, current_number);
 		}
 
-		fn on_initialize(now: T::BlockNumber) -> Weight {
-			let author = frame_system::Module::<T>::digest()
-				.logs
-				.iter()
-				.filter_map(|s| s.as_pre_runtime())
-				.filter_map(|(id, mut data)| if id == POW_ENGINE_ID {
-					T::AccountId::decode(&mut data).ok()
-				} else {
-					None
-				})
-				.next();
-
-			if let Some(author) = author {
-				<Self as Store>::Author::put(author);
-			}
-
-			T::WeightInfo::on_initialize().saturating_add(T::WeightInfo::on_finalize())
-		}
-
-		fn on_finalize(now: T::BlockNumber) {
-			if let Some(author) = <Self as Store>::Author::get() {
-				let reward = Reward::<T>::get();
-				Self::do_reward(&author, reward, now);
-			}
-
-			<Self as Store>::Author::kill();
-			<Self as Store>::AuthorDonation::kill();
+		#[weight = T::WeightInfo::set_reward_curve(curve.len() as u32)]
+		fn set_reward_curve(origin, curve: Vec<RewardPoint<T::BlockNumber, BalanceOf<T>>>) {
```

Comment:
@shawntabrizi I think we should check here that all points on the reward curve are valid. Then the `UpdateFailed` event can be removed.

Reply:
@sorpaas It only makes sense to do that check if we also control the taxation with this curve. Since ultimately, validity of the reward is based on the taxation amount and the reward amount.

Reply:
i.e. we can create a curve that is valid now, change taxation, and it would fail to update then.